Frontend/src/views/InstitutionPage/index.jsx

```diff
@@ -24,31 +25,32 @@ function InstitutionPage() {
     return (institution.adminIds || []).includes(user._id);
   }
 
-  const handleDescriptionChange = (event) => {
-    setInstitution({
-      ...institution,
-      description: event.target.value,
-    });
-  };
+  function updateInstitution() {
+    InstitutionService.getInstitution(id)
+      .then(setInstitution)
+      .catch((ignored) => { console.error(ignored); });
+  }
 
-  const handleCountryChange = (event) => {
-    setInstitution({
-      ...institution,
-      country: event.target.value,
-    });
-  };
+  function editDetails() {
+    // eslint-disable-next-line max-len
+    const details = { description: institution.description };
+    const jsonDetails = JSON.stringify(details);
+    InstitutionService.updateDetails(id, jsonDetails);
+  }
 
-  const handleNameChange = (event) => {
+  const handleDescriptionChange = (event) => {
     setInstitution({
       ...institution,
-      name: event.target.value,
+      description: event.target.value,
     });
   };
 
-  useEffect(async () => {
-    console.log(await InstitutionService.getInstitution(id));
-    setInstitution(await InstitutionService.getInstitution(id));
-    setInstitutionLoaded(true);
+  useEffect(() => {
+    InstitutionService.getInstitution(id)
+      .then((newInstitution) => {
+        setInstitution(newInstitution);
+        setInstitutionLoaded(true);
+      });
```

Comment:
What if the request fails? Then you will get the Spinner forever. We have done it differently but you can just add a .finally(()=> setInstitutionLoaded(true)) & also remove it from .then clause